src/Mapping.php

```diff
@@ -573,11 +573,11 @@ private function getBaseData(array $data)
      */
     private function buildColumns()
     {
-        $columns = $this->definition->getPrimaryKey();
+        $columns = $this->prefixTableNameTo($this->definition->getPrimaryKey());
         $required = array_merge($this->definition->getColumns(), $this->columns);
 
         foreach ($this->definition->getProperties() as $item) {
-            $required[] = $item->getLocalColumn();
+            $required[] = in_array($item->getLocalColumn(), $this->definition->getPrimaryKey()) ? $this->prefixTableNameTo($item->getLocalColumn()) : $item->getLocalColumn();
```

Comment:
@CoreyRDean is the logic in this ternary necessary, given that line 585 is prefixing all unprefixed columns with the definition's table anyway (effectively the same logic as `prefixTableNameTo()`)?